### PR TITLE
feat: 🚀 新增husky插件，支持husky、commitlint、lint-staged功能(#215)

### PR DESCRIPTION
**Pull request by nanyishixiong**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Test related changes
- [ ] Security fix
- [ ] Dependency update
- [ ] Revert changes
- [ ] Other... Please describe:

## What is the current behavior?

- Related to https://github.com/xun082/create-neat/issues/215

## What is the new behavior?

此pr新增一个husky插件，用于支持husky、commitlint、lint-staged等功能

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/@plugin/plugin-husky/generator/template/commitlint.config.cjs [hunk 1]
```diff
@@ -0,0 +1,158 @@
+// @see: https://cz-git.qbenben.com/zh/guide
```
Comment (uaenaTzx) [inline packages/@plugin/plugin-husky/generator/template/commitlint.config.cjs:1]: 这个文件默认不需要配置那么多的，要考虑到初学者用户，这些配置反而造成复杂度和心智负担

File: packages/@plugin/plugin-husky/generator/template/commitlint.config.cjs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+/** @type {import('cz-git').UserConfig} */
```
Comment (uaenaTzx) [inline packages/@plugin/plugin-husky/generator/template/commitlint.config.cjs:2]: 插入用户路径的文件后缀用 cjs 不太好，用户应该默认 js 的

File: packages/@plugin/plugin-husky/generator/index.cjs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+module.exports = (generatorAPI) => {
+  generatorAPI.extendPackage({
+    scripts: {
+      prepare: "husky",
+      commit: "git-cz",
+      format: 'prettier --write "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}"',
+      lint: 'eslint "**/*.{ts,tsx,js,jsx}" --fix',
```
Comment (uaenaTzx) [inline packages/@plugin/plugin-husky/generator/index.cjs:7]: husky 不应该有 script，就只要 lint-stage 就行了

File: packages/@plugin/plugin-husky/generator/index.cjs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        path: "node_modules/cz-git",
+      },
+    },
+    "lint-staged": {
+      "*.{ts,tsx,js,jsx}": ["npm run format", "npm run lint"],
+    },
+    devDependencies: {
+      husky: "^9.0.11",
+      "lint-staged": "^15.2.0",
+      "cz-git": "^1.7.1",
```
Comment (uaenaTzx) [inline packages/@plugin/plugin-husky/generator/index.cjs:20]: 这些依赖也确认一下，像 cz-git 这些，是没必要的功能

**Review by uaenaTzx — APPROVED** [review]

**Comment by uaenaTzx** [comment]
做好插件后本地用 create-neat 创建一个项目试试 husky 的效果

**Review by xun082 — APPROVED** [review]

**Review by xun082 — APPROVED** [review]

**Pull request merged**